### PR TITLE
initial mci support for petabank

### PR DESCRIPTION
**Pull request by kenthua**
need to validate approach

retrofitted [terraform asm module](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/master/modules/asm) for hub registration

only support added for petabank
prod-us-central1 provisioned as the MCI configuration cluster

cleanup TBD, can wait on next iteration

### CODE REVIEW

**Comment by viglesiasce** [comment]
Nice!!

File: 1_clusters/clusters.tf [hunk 1]
```diff
@@ -94,4 +97,35 @@ module "anthos-platform-prod-east" {
   subnetwork        = data.google_compute_subnetwork.anthos-platform-east1.name
   ip_range_pods     = "anthos-platform-pods-prod"
   ip_range_services = "anthos-platform-services-prod"
+  release_channel   = "STABLE"
 }
+
+resource "google_service_account" "gke_hub_sa" {
```
Comment (morgante) [inline 1_clusters/clusters.tf:103]: We could probably put this in the hub module?

File: 1_clusters/enable_mci.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    exit 1
+fi
+
+PROJECT_ID=$1
+CLUSTER_NAMES=$2
+
+# IFS if we want to handle comma delimited list of clusters
+IFS="," 
+for CLUSTER_NAME in ${CLUSTER_NAMES}; do
+  gcloud alpha container hub ingress enable \
```
Comment (morgante) [inline 1_clusters/enable_mci.sh:29]: I'm wondering if this capability should be included in the hub module?

File: 1_clusters/modules/hub-registration/main.tf [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  gke_hub_sa_key = google_service_account_key.gke_hub_key.private_key
+}
+
+resource "google_service_account_key" "gke_hub_key" {
+  service_account_id = var.gke_hub_sa
+}
+
+module "gke_hub_registration" {
+  source  = "terraform-google-modules/gcloud/google"
+  version = "~> 1.2"
```
Comment (morgante) [inline 1_clusters/modules/hub-registration/main.tf:27]: Let's update to 2.0.

File: 1_clusters/modules/hub-registration/main.tf [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  
+  platform           = "linux"
+  gcloud_sdk_version = var.gcloud_sdk_version
+  skip_download      = var.skip_gcloud_download
```
Comment (morgante) [inline 1_clusters/modules/hub-registration/main.tf:31]: We can omit this.

File: 1_clusters/modules/hub-registration/outputs.tf [hunk 5]
```diff
@@ -0,0 +1,15 @@
+/**
```
Comment (morgante) [inline 1_clusters/modules/hub-registration/outputs.tf:1]: This file can be excluded if there are none.

File: 1_clusters/modules/hub-registration/variables.tf [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  description = "The project in which the resource belongs."
+  type        = string
+}
+
+variable "location" {
+  description = "The location (zone or region) this cluster has been created in."
+  type        = string
+}
+
+variable "skip_gcloud_download" {
```
Comment (morgante) [inline 1_clusters/modules/hub-registration/variables.tf:37]: We're trying to not use this variable any more. https://github.com/terraform-google-modules/terraform-google-gcloud/blob/master/CHANGELOG.md#-breaking-changes

File: 1_clusters/modules/hub-registration/variables.tf [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  description = "Whether to skip downloading gcloud (assumes gcloud and kubectl already available outside the module)"
+  type        = bool
+  default     = true
+}
+
+variable "gcloud_sdk_version" {
```
Comment (morgante) [inline 1_clusters/modules/hub-registration/variables.tf:43]: Is there really a need to make this configurable?

File: 1_clusters/clusters.tf (hunk 1 above)
Comment (kenthua) [reply 1_clusters/clusters.tf:103]: i had it there, but i didn't want it repeating for each cluster, so there would be more logic needed if in hub or a variable, create new SA boolean

File: 1_clusters/enable_mci.sh (hunk 2 above)
Comment (kenthua) [reply 1_clusters/enable_mci.sh:29]: for MCI, this is specific to only the configuration cluster, it wouldn't apply to both clusters

File: 1_clusters/modules/hub-registration/variables.tf (hunk 7 above)
Comment (kenthua) [reply 1_clusters/modules/hub-registration/variables.tf:43]: i'll need to test this based to use the base 293

File: 1_clusters/modules/hub-registration/main.tf (hunk 3 above)
Comment (kenthua) [reply 1_clusters/modules/hub-registration/main.tf:27]: ack

File: 1_clusters/modules/hub-registration/variables.tf (hunk 7 above)
Comment (morgante) [reply 1_clusters/modules/hub-registration/variables.tf:43]: We can set a version in the module without making it configurable though.

Comment (kenthua) [reply 1_clusters/modules/hub-registration/variables.tf:43]: ah right, but if we can get by with 293, all for it

File: 1_clusters/modules/hub-registration/main.tf (hunk 4 above)
Comment (kenthua) [reply 1_clusters/modules/hub-registration/main.tf:31]: ack

File: 1_clusters/enable_mci.sh (hunk 2 above)
Comment (morgante) [reply 1_clusters/enable_mci.sh:29]: Interesting. It still seems like something we'd eventually want to add to the module, but I'm fine as is.

**Pull request merged**